tasks/main.yml

```
@@ -10,20 +10,29 @@
     use: "{{ (__snapshot_is_ostree | d(false)) |
               ternary('ansible.posix.rhel_rpm_ostree', omit) }}"
 
-- name: Show snapshot command
-  debug:
-    var: __snapshot_cmd
-    verbosity: 2
-
-- name: Run snapshot command and handle errors
+- name: Run snapshot module and handle errors
   block:
-    - name: Run snapshot command {{ snapshot_lvm_action }}
-      ansible.builtin.script: "{{ __snapshot_cmd }}"
-      args:
-        executable: "{{ __snapshot_python }}"
+    - name: Run snapshot module {{ snapshot_lvm_action }}
+      snapshot:
+        snapshot_lvm_action: "{{ snapshot_lvm_action }}"
+        ansible_check_mode: "{{ ansible_check_mode | d(false) }}"
+        snapshot_lvm_all_vgs: "{{ snapshot_lvm_all_vgs | d(false) }}"
+        snapshot_lvm_verify_only: "{{ snapshot_lvm_verify_only | d(false) }}"
+        snapshot_lvm_mount_origin: "{{ snapshot_lvm_mount_origin | d(false) }}"
+        snapshot_lvm_mountpoint_create: >-
+          "{{ snapshot_lvm_mountpoint_create | d(false) }}"
+        snapshot_lvm_unmount_all: "{{ snapshot_lvm_unmount_all | d(false) }}"
+        snapshot_lvm_percent_space_required:
+          "{{ snapshot_lvm_percent_space_required | d(omit) }}"
+        snapshot_lvm_vg: "{{ snapshot_lvm_vg | d(omit) }}"
+        snapshot_lvm_lv: "{{ snapshot_lvm_lv | d(omit) }}"
+        snapshot_lvm_snapset_name: "{{ snapshot_lvm_snapset_name | d(omit) }}"
+        snapshot_lvm_mount_options: "{{ snapshot_lvm_mount_options | d(omit) }}"
+        snapshot_lvm_fstype: "{{ snapshot_lvm_fstype | d(omit) }}"
+        snapshot_lvm_mountpoint: "{{ snapshot_lvm_mountpoint | d(omit) }}"
+        snapshot_lvm_set: "{{ snapshot_lvm_set | to_json }}"
+        snapshot_lvm_vg_include: "{{ snapshot_lvm_vg_include | d(false) }}"
       register: snapshot_cmd_raw
```

```suggestion
      register: snapshot_cmd
```
I don't think you need to convert `snapshot_cmd_raw` to `snapshot_cmd` because the module should not have the problem with "warning: setlocale: LC_ALL: cannot change locale" in the output

---

Ok.  There is a reference to snapshot_cmd_raw in verify-role-failed.yml.  I'll update the reference and the comments about why we should implement a module.